### ISSUE TITLE
This repo needs structure

### ISSUE THREAD

**Issue by j0rdsta** [why-el/crowd-written-novel#6]
I love this idea, but I think it could turn into an unmitigated disaster if there isn't some sort of boundaries put into place. It might turn into a complete circlejerk where the first letter of each sentence corresponds to some expletive (à la [James May's hidden message](https://en.wikipedia.org/wiki/James_May#Dismissal_from_Autocar)). Although it's a fun idea, nobody ain't gonna take this thing seriously!

It might be an idea to integrate the following:

### Brainstorming

I admit I have no idea about this - what does a typical writer do when thinking up ideas for a novel? Character bios, locations, basic structure of how the novel will play out? Either way, I think some brainstorming needs to be done before any actual writing is done.

Maybe some online forum like the [/r/writing](http://reddit.com/r/writing) subreddit could help.

### Ideal Length of novel, chapters

We should maintain that the novel itself should have a set maximum length (again, probably by sourcing from). Maybe it would be an idea to split the repository into one .md file for each chapter, and impose max lengths on the chapters too.

What are people's thoughts on this?

**Comment by j0rdsta** [comment]
[charlesism](https://news.ycombinator.com/item?id=9522327) on Hacker News shares a similar opinion:

> Novels written by groups are generally unreadable. This is because they have no focus, and because the immediate cheap laugh always wins. It's a shame, because with a little more forethought, such shortcomings could be fixed. Ie: start by communally developing an outline, main characters, etc. It would also be interesting to use Mechanical Turk in a systemic way to have "the crowd" determine which of several proposed plot-points is most consistent with the whole, etc.

**Comment by why-el** [comment]
@j0rdsta this is exactly the feedback and the conversation I want to start. I had structure initially but I wanted to see what the community can conjure up, and it seems we are off on a good start.

**Comment by Ghoughpteighbteau** [comment]
At the very minimum, one sentence is causing problems, accept pull requests will multiple lines.

**Comment by dav-** [comment]
:+1: 
I think it's essential for an outline to be written before the actual substance of each section. The direction of the story would still be determined by the community, but it would allow for a system of judging whether a PR should be merged or not. 

For example, if the outline states that chapter 1 flows as such: 

`bob bakes cake -> bob goes to market -> bob sells cake`

If someone submits a pull request involving aliens destroying New York, it would probably not fit the spec. 

I'm not sure how the outline would best be developed, perhaps some type of repeating sequence of *"suggestions"*, *"voting on suggestions"*, *"approval and refinement"*.

**Comment by jfonte** [comment]
Outline? How's this premise: A man disappears. His concerned wife walk into a police station and is greeted by strange policeman who gives her a strange handout and sends her on her way. From here we go back a few days leading into the man's disappearance. Vibe/mood should be Gilliam/Lynch-esque

**Comment by dav-** [comment]
@jfonte I'm not sure if this would be the best place for suggestions like that, I think we should first come up with a system for creating the outline.

**Comment by jfonte** [comment]
Then I think we're going to need some kind of voting system, no?

**Comment by dav-** [comment]
@jfonte Any ideas on how to implement that?

**Comment by n-r** [comment]
The obvious solution is to use the comments to debate for and against changes.

**Comment by jfonte** [comment]
@dav- GitPoll maybe? http://poll.gitrun.com

**Comment by jfonte** [comment]
@n-r Debating with every participant won't be practical without  voting

**Comment by n-r** [comment]
You are correct about that. Ideally, would you like to have a list of all pull requests and have everyone vote for the best request, one account, one vote? I feel like that would favor older pull requests over newer ones. Also, when would a pull request be sufficiently good to be added? Would the best pull request, for example, be added every fifteen minutes?

**Comment by jfonte** [comment]
@n-r (1) Yes, it would probably favor old requests. It would be a trade-off (reddit suffers from this, no?) (2)  Time based voting would probably be the way to go. Maybe we wait at least 24-48hrs for pull requests. A week if we are really going crazy (3) We should have a minimum word count

**Comment by Ghoughpteighbteau** [comment]
Personally? And to put it crudely as possible, for now I say we just throw shit at the wall and see what sticks. We can organize the mess later. 

If you put too much credence and structure in voting, you almost go on record stating "This is what we've all agreed, see, here's the vote, there's no going back, not without another vote". The thing is, everything is easily undone, so maybe something more organic is appropriate here.

**Comment by jfonte** [comment]
What about the model HitRecord (Gordon Levitt's project) uses? That seems to work organically.

**Comment by dav-** [comment]
@Ghoughpteighbteau I feel like the issue with that is that merging a single PR will completely invalidate many others just because of a change in direction in the story. If 10 people spend 15 minutes each on their submissions, at most two or three will be compatible with each other. All of a sudden 70-80% of the work becomes essentially wasted effort.

**Comment by dav-** [comment]
@jfonte That looks interesting - can you link to something explaining how their collaboration system works?

**Comment by Ghoughpteighbteau** [comment]
yah I'm not familiar with that system jfonte.

**Comment by jfonte** [comment]
@dav- @Ghoughpteighbteau https://www.youtube.com/watch?v=u6fqJt3mm6I Best way to experience it, is to join at HitRecord.org

**Comment by n-r** [comment]
@jfonte The difference that I see between HitRecord and a novel is scale.

The pieces of art on HitRecord seem small enough that someone can take them in all by (him/her)self and make another related piece of art. An entire novel may be hard for a new contributor to absorb entirely. Furthermore, it would be even more difficult for someone to make respond with a novel-length or greater work based on the novel. Not that it can't be done, but I'm not sure the contributors would be dedicated enough to do so.

What I said may be incorrect--all of my knowlege about HitRecord comes from watching a few of their introductory videos.

Back to an earlier topic, I agree with @jfonte about the importance of a minimum word count. Without it, the novel becomes too fragmented.

**Comment by jfonte** [comment]
@n-r we can start with a short story and then scale it into a novel :) No need to go crazy. We start small. 5k word project to start perhaps?

**Comment by Ghoughpteighbteau** [comment]
yes! That's what I think. Where are you @why-el 
Roll the dice on a pull request. Take all the ones you invalidated and put them in a big idea bin we can pull from an add to  the main story. Expand expand expand! and for the love of god. Follow #16!

**Comment by dav-** [comment]
@Ghoughpteighbteau I think @why-el went to sleep :sob:.

**Comment by jfonte** [comment]
@dav- I think we all need to go get some shut eye. It's 5:45am over here in Germany...

**Comment by why-el** [comment]
Yep, I did. I will process this after work and see what we can come up with. Great discussion so far.

**Comment by Ghoughpteighbteau** [comment]
re voting. Maybe the the subreddit #28 could be used? I don't too much like the idea of having to create another account to work on the project, but seeing as I already got one :3

**Comment by Ghoughpteighbteau** [comment]
One of the guys at Hacker News proposed an organic darwinian structure that I believe has merrit. Every person that contributes to this repository does so on their own branch. Everyone reads through each branch and inherently commits to that branch. Merges can happen, but must be controlled and form their own branch. Branches that get ignored are languished and pruned.

This solves a majority of the problems, and allows everyone to work more or less independently. Branches that lose validity can still be recycled for their ideas. 

To my mind, this gives the project the greatest chance of success.

**Comment by n-r** [comment]
Can you post a link to that comment?

**Comment by Ghoughpteighbteau** [comment]
yah. https://news.ycombinator.com/reply?id=9522580&goto=item%3Fid%3D9522106

**Comment by n-r** [comment]
I like that idea, but there has to be some sort of list of all branches. Otherwise, there would be no incentive to make a branch because no one would know about it and thus no one would contribute. It would be a list of comments without replies, a tree with all leaves directly connected to the root, if you will.

The wiki is a possible way to solve this problem

**Comment by Ghoughpteighbteau** [comment]
Well I don't agree with that.

github provides a search-able list of branches.
![here](http://i.imgur.com/MNkmFZs.png)

and you can quickly review branches and their progeny with local tools like gitk
![as an example](http://i.imgur.com/nujAahd.png)

and you can simply just grab the list of current branches locally by typing `git branch` on the CLI.

Truth be told, I'm not actually sure how to achieve this kind of structure using a wiki. I can make some guesses at it, but the strategies I'm coming up with feel awkward.

**Comment by why-el** [comment]
What do you guys think of what @tanishalfelven in this PR: https://github.com/why-el/crowd-written-novel/pull/29 ? 

I think it's a good start, though it does lock everything to a per-chapter workflow and it's essentially just voting.

**Comment by n-r** [comment]
Sorry, @Ghoughpteighbteau, I was confusing forks with branches. Branches seem like an awesome idea, and I am in full support of them.

**Comment by Ghoughpteighbteau** [comment]
Kinda meh. but I see accepting pull requests as important, and not getting married to an idea because it gets brought in. The delete key is just a button, after all.

**Comment by n-r** [comment]
The thing that I like about having multiple branches is that it is non-destructive. Maybe someone writes a scene about Bill having dinner with his family and another person writes a scene about an encounter between Bill and space aliens. If one scene fits better in the story line at that place, it can be merged directly into master. However, the other scene doesn't need to be forgotten--it can be merged back into the story line at a later point.

Additionally, using branches would allow pull requests to be accepted more easily. A commit on an individual's branch would not need wider verification to be added. The whole community would only need to add input when a branch is being merged with master.

**Comment by Ghoughpteighbteau** [comment]
no master!

If a master exists people will commit to it.

**Comment by n-r** [comment]
I think that a master branch curated by @why-el (or mods, somehow) would be a great idea, and every pull request by unqualified people that commits to it would get denied (with a helpful message). A master branch would give a good reference point for a new contributor.

**Comment by daslicious** [comment]
I have heard writers reference 'Story' by Robert McKee as very helpful for understanding the requirements/components of a story.

**Comment by jfonte** [comment]
@daslicious one can make a story in 3 words or a sentence. What's hard is telling a compelling story. If we're going to go the route of teaching people how to tell a story, hell, I'd suggest having a reading list that included titles like Campbell's "The Hero with a Thousand Faces", King's "On Writing" and "The Elements of Style" FOR A START.

**Comment by jfonte** [comment]
Can anyone come up with a small graphic of what our workflow should be?

**Comment by n-r** [comment]
In my mind:

A Contributor
-----------------
1. clone the repo
1. make a branch
1. change that branch
1. submit a pull request

Another Contributor
--------------------------
1. clone the repo
1. find two or more compatible branches
1. merge those branches together
1. submit a pull request

Head of the Repo / Mods
--------------------------------
1. wait a few hours
1. find a promising branch
1. merge it into master
1. commit

**Comment by daslicious** [comment]
@jfonte Yeah, that's what I'm suggesting.  Seems a good place to start.  There's obviously tons to be gained from non-instructional books and a short list of those might not be a bad idea.

**Comment by n-r** [comment]
Maybe we could have a separate file containing writing tips for the project? That could help new contributors and aid consistancy in the novel's style.

**Comment by daslicious** [comment]
@n-r +1

**Comment by jfonte** [comment]
Would it make more sense to contribute in paragraphs with a min/max length instead of just fragmented sentences?

**Comment by n-r** [comment]
@jfonte Maybe we contribute stand-alone scenes ("episodes")? Each could be two or three paragraphs long--nothing major, but something substantive. They could then be expanded later so as to not sound choppy.

**Comment by jfonte** [comment]
I thought about it and came to the conclusion that this will never work out (at least for me). It's already a lot of work for one or even two people collaborating, to deal with editing and proofreading. I really don't see this scaling to X collaborators. Sorry mates. Best of luck though.

**Comment by why-el** [comment]
Thanks Josh for the words. I get caught up with life over the weekend, but I am going over this tonight and deciding who can help me.

**Comment by why-el** [comment]
Ok, from my readings it seems that the consensus is on using branches and having collaborators pick branches they lick and slice and match. There is also consensus over having some word limits. 

If this is what you guys mean, I will take a look at the PRs, merge some, and leave some open for other people to build on.

I will also update the README to explain the new process.
